yum/artifact-registry.py

```diff
@@ -0,0 +1,33 @@
+import google.auth
+import google.auth.transport.requests
+from google.auth.exceptions import DefaultCredentialsError, RefreshError
+
+from yum.plugins import TYPE_CORE
+
+requires_api_version = '2.3'
+plugin_type = (TYPE_CORE,)
+
+
+def prereposetup_hook(conduit):
+  conduit.info(2, 'Enabling Artifact Registry authentication')
+  for repo in conduit.getRepos().listEnabled():
+    for url in repo.urls:
+      if 'pkg.dev' in url:
+        _add_headers(repo)
+        break  # Stop looking at URLs
```

Comment:
I think it's a bit cleaner to use `any` here.

```
if any('pkg.dev' in url for url in repo.urls):
  _add_headers(repo)
```

Reply:
mine short circuits quicker